packages/components/src/components/domainproperties/dataset/DatasetPropertiesPanelFormElements.tsx

```diff
@@ -0,0 +1,297 @@
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {Col, Radio, Row} from 'react-bootstrap';
+import {DatasetModel} from './models';
+import {LabelHelpTip, SelectInput} from "../../..";
+import {Creatable} from 'react-select'
+import {fetchCategories} from "./actions";
+import {DatasetSettingsInput} from "./DatasetPropertiesAdvancedSettings";
+import {CheckBox} from "../list/ListPropertiesPanelFormElements";
+import "../../../theme/dataset.scss";
+
+interface BasicPropertiesInputsProps {
+    model?: DatasetModel;
+    onInputChange: (any) => void;
+    name?: string;
+    description?: string;
+    categoryId?: number;
+    onCategoryChange?: any;
+    label?: string;
+}
+
+interface BasicPropertiesTitleProps {
+    title: string
+}
+
+export class BasicPropertiesTitle extends React.PureComponent<BasicPropertiesTitleProps> {
+    render() {
+        return <div className="domain-field-section-heading">{this.props.title}</div>;
+    }
+}
+
+export class DescriptionInput extends React.PureComponent<BasicPropertiesInputsProps> {
+    render() {
+        const { description, onInputChange } = this.props;
+        const value = description === null ? '' : description;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4}>
+                    Description
+                </Col>
+
+                <Col xs={7}>
+                    <textarea
+                        className="form-control textarea-noresize"
+                        id="description"
+                        value={value}
+                        onChange={onInputChange}
+                    />
+                </Col>
+            </Row>
+        );
+    }
+}
+
+export class SelectPropertyInput extends React.PureComponent<any> {
+    render() {
+        const { label, labelKey, valueKey, onInputChange, options, disabled } = this.props;
+
+        return(
+            <Row className={'margin-top'}>
+                <Col xs={4} >
+                    {label}
+                </Col>
+
+                <Col xs={7} >
+                    <SelectInput
+                        onChange={onInputChange}
+                        options={options}
+                        inputClass="" // This attr is necessary for proper styling
+                        labelClass=""
+                        valueKey={valueKey}
+                        labelKey={labelKey}
+                        formsy={false}
+                        multiple={false}
+                        required={false}
+                        disabled={disabled}
+                    />
+                </Col>
+
+                <Col lg={3}/>
+            </Row>
+        );
+    }
+}
+
+export class BasicPropertiesFields extends React.PureComponent<BasicPropertiesInputsProps, any> {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+             category: this.props.model.categoryId,
```

Comment:
I don't think you want category in state, but instead should just be using it from the model prop directly.